forgevtt-module.js

```diff
@@ -1872,80 +1879,95 @@ class ForgeVTT_FilePicker extends FilePicker {
                 }
                 // The only bucket with no token is our own assets library
                 if (!bucket.jwt) {
-                    return ["forgevtt", target, bucket.key];
+                    return ["forgevtt", forgePath, bucket.key];
                 }
                 const info = ForgeAPI._tokenToInfo(bucket.jwt);
                 // Get the key's root dir and trim the leading slash.
                 const rootDir = info.keyOptions?.assets?.rootDir?.replace(/^\/+/, "");
-                if (rootDir && target.startsWith(rootDir)) {
-                    target = target.slice(rootDir.length);
-                    return ["forgevtt", target, bucket.key];
+                if (rootDir && forgePath.startsWith(rootDir)) {
+                    const bucketPath = forgePath.slice(rootDir.length);
+                    return ["forgevtt", bucketPath, bucket.key];
                 }
                 // Old custom API keys do not have a root dir, so if the user id matches the target
                 // is in here somewhere.
                 if (!rootDir) {
-                    return ["forgevtt", target, bucket.key];
+                    return ["forgevtt", forgePath, bucket.key];
                 }
```

Comment:
I must be missing something then, as far as I understand the initial implementation:
```js
for (const bucket of this._forgeBucketIndex) {
  if (bucket.userId !== userId) {
    // Only "my-assets" bucket matches my `userId`, so we skip any other bucket
    continue; 
  }
  // Only "my-assets" bucket reaches the following statements
  if (!bucket.jwt) {
    // "my-assets" doesn't have a JWT, so this will always be the outcome of the `for` loop
    return ["forgevtt", target, bucket.key];
  }
  // No bucket ever reaches the following statements
  const info = ForgeAPI._tokenToInfo(bucket.jwt);
  ...
}
```
:thinking:

Reply:
`userId` here is the user ID from the asset URL, not the current Forge user. "my-assets" would only get past that first `if` when the asset is one of the user's own.

The actual reason the select was always defaulting to the first option was because this method was still returning the `bucket.key` as the third value, when v12 expects the index instead. You've since fixed that, but I'll note that you're still effectively using `userId` as the original logic did. 😉 `sharedBuckets.find((bucket) => userId === bucket.userId)` is the same as `for (const bucket...) { if (bucket.userId !== userId) continue; }`